### PR TITLE
Add mm_get_heap_free_size()

### PR DESCRIPTION
**Pull request by jeongchanKim**

### CODE REVIEW

**Review by abhishek-samsung — COMMENTED** [review]
LGTM

File: os/mm/mm_heap/mm_getsize.c [hunk 1]
```diff
@@ -90,3 +90,37 @@ size_t mm_get_largest_freenode_size(void)
...
+		target_heap = &heap[heap_idx];
+		free_size += (target_heap->mm_heapsize - target_heap->total_alloc_size);
+	}
+#else
+	/* User can only have single heap. */
+	heap = BASE_HEAP;
+	if (!heap) {
+		mdbg("Fail to find the heap.\n");
+		return 0;
+	}
```
Comment (sunghan-chang) [inline os/mm/mm_heap/mm_getsize.c:122]: Is it possible to be null?

File: os/mm/mm_heap/mm_getsize.c [hunk 2]
```diff
@@ -90,3 +90,37 @@ size_t mm_get_largest_freenode_size(void)
...
+	size_t free_size = 0;
+	int heap_idx;
+	heap = g_kmmheap;
+	struct mm_heap_s *target_heap;
+
+	/* Kernel can have multi-heap, so traversing all n-heaps and add all free sizes. */
+	for (heap_idx = HEAP_START_IDX; heap_idx <= HEAP_END_IDX; heap_idx++) {
+		target_heap = &heap[heap_idx];
+		free_size += (target_heap->mm_heapsize - target_heap->total_alloc_size);
+	}
```
Comment (sunghan-chang) [inline os/mm/mm_heap/mm_getsize.c:115]: There is no return value in this #ifdef.

Comment (jeongchanKim) [reply os/mm/mm_heap/mm_getsize.c:111]: Added. Thanks.

File: os/mm/mm_heap/mm_getsize.c [hunk 3]
```diff
@@ -90,3 +90,37 @@ size_t mm_get_largest_freenode_size(void)
...
+#else
+	/* User can only have single heap. */
+	heap = BASE_HEAP;
+	if (!heap) {
+		mdbg("Fail to find the heap.\n");
```
Comment (kishore-sn) [inline os/mm/mm_heap/mm_getsize.c:120]: Can you pls change the debug statement to give more information?
Like "Failed to retrieve the heap region for xxx application".
Also at line 86.

File: os/mm/mm_heap/mm_getsize.c [hunk 4]
```diff
@@ -90,3 +90,37 @@ size_t mm_get_largest_freenode_size(void)
...
 }
+
+/****************************************************************************
+ * Name: mm_get_heap_free_size
+ *
+ * Description:
+ *   returns free size in the heap
+ ****************************************************************************/
+#ifdef CONFIG_DEBUG_MM_HEAPINFO
+size_t mm_get_heap_free_size(void)
```
Comment (kishore-sn) [inline os/mm/mm_heap/mm_getsize.c:101]: I think we can make these as inline functions.

File: os/mm/mm_heap/mm_getsize.c (hunk 3 above)
Comment (jeongchanKim) [reply os/mm/mm_heap/mm_getsize.c]: Modified. Thanks.

File: os/mm/mm_heap/mm_getsize.c (hunk 4 above)
Comment (jeongchanKim) [reply os/mm/mm_heap/mm_getsize.c:109]: I will do refactoring most of the functions in `mm_getheap.c` and `mm_getsize.c` about separating umm and kmm.
I will change it at that time.
Thanks:)

File: os/mm/mm_heap/mm_getsize.c (hunk 2 above)
Comment (sunghan-chang) [reply os/mm/mm_heap/mm_getsize.c:123]: Not added. Please check it again.

**Review by sunghan-chang — APPROVED** [review]

File: os/mm/mm_heap/mm_getsize.c (hunk 2 above)
Comment (jeongchanKim) [reply os/mm/mm_heap/mm_getsize.c:123]: Added. Thanks.

**Pull request merged**